textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/relation/RelationManager.java

```diff
@@ -0,0 +1,363 @@
+        if (! checkCatalogExistence()) {
+            initializeCatalog();
+        }
+    }
+
+    public static RelationManager getRelationManager() throws StorageException {
+        if (singletonRelationManager == null) {
+            synchronized (RelationManager.class) {
+                if (singletonRelationManager == null) {
+                    singletonRelationManager = new RelationManager();
+                }
+            }
+        }
+        return singletonRelationManager;
+    }
+
+    public boolean checkTableExistence(String tableName) {
+        try {
+            getTableDirectory(tableName);
+            return true;
+        } catch (StorageException e) {
+            return false;
+        }
+    }
+
+    // create a new table, tableName must be unique
+    public void createTable(String tableName, String indexDirectory, Schema schema, String luceneAnalyzer)
+            throws StorageException {
+        // table should not exist
+        if (checkTableExistence(tableName)) {
+            throw new StorageException(String.format("Table %s already exists.", tableName));
+        }
+        
+        Schema tableSchema = Utils.getSchemaWithID(schema);
+                
+        // write collection catalog
+        DataStore collectionCatalogStore = new DataStore(CatalogConstants.COLLECTION_CATALOG_DIRECTORY,
+                CatalogConstants.COLLECTION_CATALOG_SCHEMA);      
+        insertTupleToDirectory(collectionCatalogStore, LuceneAnalyzerConstants.getStandardAnalyzer(),
+                CatalogConstants.getCollectionCatalogTuple(tableName, indexDirectory, luceneAnalyzer));
+       
+        // write schema catalog
+        DataStore schemaCatalogStore = new DataStore(CatalogConstants.SCHEMA_CATALOG_DIRECTORY,
+                CatalogConstants.SCHEMA_CATALOG_SCHEMA);
+        for (ITuple tuple : CatalogConstants.getSchemaCatalogTuples(tableName, tableSchema)) {
+            insertTupleToDirectory(schemaCatalogStore, LuceneAnalyzerConstants.getStandardAnalyzer(), tuple);
+        }
+        
+    }
+
+    // drop a table
+    public void deleteTable(String tableName) throws StorageException {
+        Query catalogTableNameQuery = new TermQuery(new Term(CatalogConstants.COLLECTION_NAME, tableName));
+        
+        DataWriter collectionCatalogWriter = new DataWriter(CatalogConstants.COLLECTION_CATALOG_DATASTORE, 
+                LuceneAnalyzerConstants.getStandardAnalyzer());
+        collectionCatalogWriter.deleteTuple(catalogTableNameQuery);
+        
+        DataWriter schemaCatalogWriter = new DataWriter(CatalogConstants.SCHEMA_CATALOG_DATASTORE,
+                LuceneAnalyzerConstants.getStandardAnalyzer());
+        schemaCatalogWriter.deleteTuple(catalogTableNameQuery);
+    }
+
+    // insert a tuple to a table, returns the ID field
+    public IDField insertTuple(String tableName, ITuple tuple) throws StorageException {
+        if (! checkTableExistence(tableName)) {
```

Comment:
Split `insertTuple` to two functions, one external for users to call (where `_id` cannot be present) and one internal for our own insert (where `_id` has to be present).

Also add a comment to the codebase to explain that the engine does not enforce uniqueness of any attribute.